engine/src/main/scala/cromwell/engine/backend/jes/JesEngineFunctions.scala

```diff
@@ -1,20 +1,26 @@
 package cromwell.engine.backend.jes
 
-import java.nio.file.FileSystem
+import java.nio.file.{FileSystem, Path}
 
 import better.files._
+import cromwell.core.{CallContext, WorkflowContext}
 import cromwell.engine.backend.io._
-import cromwell.engine.backend.{CallContext, CallEngineFunctions, WorkflowContext, WorkflowEngineFunctions}
+import cromwell.{CallEngineFunctions, WorkflowEngineFunctions}
 import wdl4s.values._
 
 import scala.language.postfixOps
 import scala.util.Try
-class JesWorkflowEngineFunctions(fileSystems: List[FileSystem], context: WorkflowContext) extends WorkflowEngineFunctions(fileSystems, context) {
+class JesWorkflowEngineFunctions(fileSystems: List[FileSystem], context: WorkflowContext) extends WorkflowEngineFunctions(context) {
+  import cromwell.core.PathFactory._
+
   override def globPath(glob: String): String = context.root.toAbsolutePath(fileSystems).resolve(JesBackend.globDirectory(glob)).toString
   override def glob(path: String, pattern: String): Seq[String] = {
     path.toAbsolutePath(fileSystems).asDirectory.glob("**/*") map { _.path.fullPath } filterNot { _.toString == path } toSeq
   }
-  override def adjustFilePath(path: String) = if (!path.isGcsUrl) context.root.toAbsolutePath(fileSystems).resolve(path).toString else path
+
+  override def adjustFilePath(path: Path) = if (!path.isAbsolute) context.root.toAbsolutePath(fileSystems).resolve(path) else path
```

Comment:
I don't think this should/can be in GCS filesystem, we're prepending the workflow root path to this path if we detect that it's not absolute. For that we need the workflow root which we don't have in GCS Filesystem.
Moreover this might actually be broken I need to double-check.